libraries/Update/src/OTA_Update.cpp

```diff
@@ -0,0 +1,70 @@
+/* Cloud task
+
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
+#include <stdio.h>
+#include <string.h>
+
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "esp32-hal-log.h"
+#include "esp_http_client.h"
+#include "esp_https_ota.h"
+
+#include"OTA_Update.h"
+#include"Esp.h"
+
+const char * server_cert;
+const char * ota_url;
+bool skip_cert_common_name_check;
+void (*http_func_ptr)(HttpEvent_t*);
+
+esp_err_t _http_event_handler(esp_http_client_event_t *evt)
+{
+    http_func_ptr(evt); 
+    return ESP_OK;
+}
+
+void aws_iot_task(void *param)
+{
+    log_i("Starting OTA example");
+    log_i("Attempting to connect to the server.........");
+
+    esp_http_client_config_t config;
+    config.url = ota_url;
+    config.cert_pem = server_cert;
+    config.skip_cert_common_name_check = skip_cert_common_name_check;
+    config.event_handler = _http_event_handler;
+    esp_err_t ret = esp_https_ota(&config);
+    if (ret == ESP_OK) {
+        ESP.restart();
+    } else {
+        log_e("Firmware upgrade failed");
+    }
+    while (1) {
```

Comment:
If update failed, maybe its better to delete task instead?

Reply:
yup. delete the task if not needed